src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,30 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+    private static final String LINE_SEPARATOR = System.lineSeparator();
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        LocalDate dateStart = LocalDate.parse(dateFrom, FORMATTER);
+        LocalDate dateFinish = LocalDate.parse(dateTo, FORMATTER);
+        StringBuilder res = new StringBuilder("Report for period ").append(dateFrom)
+                .append(" - ").append(dateTo);
+        int amountOfEmployees = names.length;
+        int[] salaries = new int[amountOfEmployees];
+        for (int i = 0; i < amountOfEmployees; i++) {
+            for (int j = 0; j < data.length; j++) {
+                String[] dataArray = data[j].split(" ");
+                LocalDate dateToCompare = LocalDate.parse(dataArray[0], FORMATTER);
+                if (data[j].contains(names[i]) && !dateToCompare.isBefore(dateStart)
+                        && !dateToCompare.isAfter(dateFinish)) {
+                    salaries[i] += Integer.parseInt(dataArray[2]) * Integer.parseInt(dataArray[3]);
```

Comment:
create constants for array indexes (e.g. DATE_INDEX, WORKING_HOURS_INDEX, INCOME_PER_HOUR_INDEX)